tests/integration/awslambda/test_lambda_integration.py

```diff
@@ -396,6 +440,308 @@ def test_python_lambda_subscribe_sns_topic(
         assert "Subject" in notification
         assert subject == notification["Subject"]
 
+    def test_event_source_mapping_with_destination_config_kinesis_stream(
+        self,
+        lambda_client,
+        create_lambda_function,
+        sqs_client,
+        sqs_queue_arn,
+        sqs_create_queue,
+        iam_create_role_with_policy,
+        kinesis_client,
+    ):
+        try:
+            function_name = f"lambda_func-{short_uid()}"
+            role = f"test-lambda-role-{short_uid()}"
+            policy_name = f"test-lambda-policy-{short_uid()}"
+            role_arn = iam_create_role_with_policy(
+                RoleName=role,
+                PolicyName=policy_name,
+                RoleDefinition=lambda_role,
+                PolicyDefinition=s3_lambda_permission,
+            )
+
+            response = create_lambda_function(
+                handler_file=TEST_LAMBDA_PYTHON,
+                func_name=function_name,
+                runtime=LAMBDA_RUNTIME_PYTHON37,
+                role=role_arn,
+                libs=TEST_LAMBDA_LIBS,
+            )
+
+            response["CreateFunctionResponse"]["FunctionArn"]
+
+            # test destination-config for create_event_source_mapping
+            kinesis_name = f"test-kinesis-{short_uid()}"
+            kinesis_client.create_stream(StreamName=kinesis_name, ShardCount=1)
+
+            result = kinesis_client.describe_stream(StreamName=kinesis_name)["StreamDescription"]
+            kinesis_arn = result["StreamARN"]
+
+            # wait for stream-status "ACTIVE"
+            status = result["StreamStatus"]
+            if status != "ACTIVE":
+
+                def check_stream_active():
+                    state = kinesis_client.describe_stream(StreamName=kinesis_name)[
+                        "StreamDescription"
+                    ]["StreamStatus"]
+                    if state != "ACTIVE":
+                        raise Exception(f"StreamStatus is {state}")
+
+                retry(check_stream_active, retries=6, sleep=1.0, sleep_before=2.0)
```

Comment:
this reminds me of the code you wrote above for waiting for a kinesis stream to become active 🙂 consider calling that function here to reduce repeated code.